roles/compliance/tasks/main.yml

```diff
@@ -1,8 +1,20 @@
 ---
-- name: Check for Insights configuration
+- name: Get the insights-client status
+  ansible.builtin.command:
+    argv:
+      - insights-client
+      - --status
+  changed_when: false
+  register: __compliance_insights_status
+  failed_when: __compliance_insights_status.rc not in [0,1]
+
+- name: Check insights-client is registered
   ansible.builtin.assert:
-    that: ansible_local.insights is defined
-    fail_msg: Unable to find insights fact. Have you installed the insights client?
+    that:
+      - >-
+        "This host is registered" in __compliance_insights_status.stdout
+        or "Registered" in __compliance_insights_status.stdout
```

Comment:
Where the "Registered" come from?

I would rather check with this [line](https://github.com/RedHatInsights/insights-core/blob/8e2fcbaa25b797e93e185f83f2afb5987b418187/insights/client/support.py#L41)
`'Insights API confirms registration.'`

Reply:
From the `rhc` role:
https://github.com/linux-system-roles/rhc/blob/e69a63a9f5d77f00c84d7a2c591f158cde7d634a/tasks/insights-client-unregistration.yml#L6-L13
as implemented by https://github.com/linux-system-roles/rhc/pull/15

Reply:
Too many prompts with the same information :laughing: